src/CSLOLSystemTrayManager.h

```diff
@@ -0,0 +1,52 @@
+#ifndef CSLOLSYSTEMTRAYMANAGER_H
+#define CSLOLSYSTEMTRAYMANAGER_H
```

Comment:
I'm not sure what's the point of adding this C++ class, qml provides SystemTrayIcon implementation: https://doc.qt.io/qt-6/qml-qt-labs-platform-systemtrayicon.html 
In fact i'm pretty sure that's what we used before

Reply:
Yes, that is what you used before. However, when you try and interact with the system tray icon when the initial visibility for the application window is set to false, trying to open the context menu immediately crashes the program. You can work around this by having the window start visible and then hiding it in the Component.onCompleted section, but this results in the window flashing open briefly on startup, where I wanted the program to start more unobtrusively. I went with this approach to separate the tray from the actual application window to prevent these behaviors, but I understand if this is too big of a change.